### ISSUE TITLE
Add 'has part' as a possible relation between 2 MFs

### ISSUE THREAD

**Issue by pgaudet** [geneontology/noctua-visual-pathway-editor#13]
Hi @tmushayahama 

We'd like to add 'has part' as a relation between two different activity units: 

<img width="624" alt="image" src="https://user-images.githubusercontent.com/4782928/159901421-76738e80-de62-4dcb-ab67-57c09478e1cb.png">

Thanks, Pascale

**Comment by tmushayahama** [comment]
@pgaudet @thomaspd can you explain more? As in how to integrate it and specs

- is it MF --has part--> MF?
- what would be the question to choose from the relationship selector?

**Comment by pgaudet** [comment]
Hi @tmushayahama 

> MF --has part--> MF?

Yes 

> what would be the question to choose from the relationship selector?

This is what I propose

<img width="892" alt="image" src="https://user-images.githubusercontent.com/4782928/160761728-187ab475-f3c4-42e2-81cb-c5c80f4a28c5.png">


@thomaspd is that OK for you?

**Comment by tmushayahama** [comment]
@thomaspd @pgaudet the only potential confusion with the latter is editing. Linking other activities in an activity detail page will make editing have cascade effect. I.e. editing/removing a subfunction which belongs to another activity will make the other activity get edited as well. Will this be a problem?

**Comment by pgaudet** [comment]
> editing/removing a subfunction which belongs to another activity will make the other activity get edited as well. Will this be a problem?

What do you mean exactly? For example if we remove the subfunction, will the the primary function change, or just its metadata (date, modified by?) I suppose it would not affect the evidences (but would behave like when we edits inputs for example)? 

Thanks, Pascale

**Comment by pgaudet** [comment]
@tmushayahama implemented this as suggested by @thomaspd 

When I opened an existing model, the relation was not shown: http://bioghost2.usc.edu:4205/workbench/noctua-visual-pathway-editor/?model_id=gomodel%3A5b528b1100000061

<img width="544" alt="image" src="https://user-images.githubusercontent.com/4782928/162404915-4a9c6961-5490-43ad-8aaf-2080d736c4c1.png">


I dont know why this is not shown?

**Comment by pgaudet** [comment]
Got it! It is shown within the main box, so we can only see it in the 'main' function box: 

<img width="226" alt="image" src="https://user-images.githubusercontent.com/4782928/162405968-95ddedfe-680c-4c66-9855-bf3fc9f996da.png">


which means we have see the information twice

<img width="549" alt="image" src="https://user-images.githubusercontent.com/4782928/162406323-06506855-78eb-42be-b31d-dbaed2924ac2.png">

**Comment by pgaudet** [comment]
It seems to me my proposition to add the link in the relations editor rather than in the Activity Unit editor was simpler - now we may want to add some filter on the 'has part' activity display. I cannot delete the other box (with cis-regulatory region sequence-specific binding), as it is technically the same one as shown within the DNA binding transcription factory activity box. 

That being said, *adding* a new subfunction works well (see 'has part DNA binding' in the NA binding transcription factory activity box.) However in the medium term we'll have a lot of imported annotations via other sources (ie not everything is edited via the Visual Editor), so we'll need to find a way to make the display clearer.

Thanks, Pascale

**Comment by thomaspd** [comment]
I think it's working just as expected. The issue is that in your model, there was an "enabled by" on the "sequence-specific DNA binding" MF, which is not necessary because it has the same enabler as the parent function that it's a part of ("transcription factor activity"). So I edited the model to remove this enabled by, and now it looks good to me. Here's the model before:
![Screenshot 2022-04-08 at 09 30 54](https://user-images.githubusercontent.com/11966328/162486116-85af0a1c-0a7e-4927-9c88-dd0767e31ccd.png)
And here's the model after:
![Screenshot 2022-04-08 at 09 37 57](https://user-images.githubusercontent.com/11966328/162486334-fa936034-ab72-4237-af63-8f1e543edf6b.png)

And here's the view in the visual pathway editor:
![Screenshot 2022-04-08 at 09 48 51](https://user-images.githubusercontent.com/11966328/162486539-9d5e8d30-8956-4640-809a-9a5acec0b84c.png)

I also checked the annotations that come out, and the seq-specific DNA binding annotation is there, even when the enabled by is removed, because I put the evidence on the has part relation instead, as is standard for all the other modifiers of the main activity. So I think we're good to go!
